### PR TITLE
Native Snap for Chrome and Safari (#9593)

### PR DESCRIPTION
**Pull request by johannes-nivus**
This is my first PR ever. And also the first time I tried to work with github. So please be patient.

Summary:
Use Native Snap where possible (Chrome and Safari)
Changed Waypoint boundaries to middle of previous element, because of Chrome snapping native to fractions of a pixel which leads to setting the value to the wrong element in a picker.
Todo:
Improve the browser conditionals (Perhaps include Versions etc.):
I haven't found a definite way how to do this throughout the code, feel free to point me in the right directions.

### CODE REVIEW

**Comment by hkollmann** [comment]
Many thanks for your first PR! Great!

With https://caniuse.com/#search=scroll-snap scroll-snap-type is supported in all modern browsers.
So what do you think about testing the support like suggested here:
https://stackoverflow.com/questions/33615761/js-check-if-browser-supports-css-snap-points

if ("scroll-snap-type" in document.body.style) {
    // do stuff with `scroll-snap-points`
  } else {
    // `scroll-snap-points` not supported
  }

**Comment by hkollmann** [comment]
Qooxdoo style for the test would be to add an static method to qx.bom.client.Css and add an new enviroment variable at the end of the file in the defer section.

This could be used in your code then.

**Review by oetiker — APPROVED** [review]

File: framework/source/class/qx/ui/mobile/container/MNativeScroll.js [hunk 1]
```diff
@@ -133,15 +136,19 @@ qx.Mixin.define("qx.ui.mobile.container.MNativeScroll",
 
       this._setCurrentX(scrollLeft);
       this._setCurrentY(scrollTop);
+      
+      if(qx.core.Environment.get("browser.name").indexOf("chrome") == -1 &&
+         qx.core.Environment.get("browser.name").indexOf("safari") == -1) {
```
Comment (oetiker) [inline framework/source/class/qx/ui/mobile/container/MNativeScroll.js]: don't repeat yourself

File: framework/source/class/qx/ui/mobile/container/MNativeScroll.js [hunk 2]
```diff
@@ -113,13 +113,16 @@ qx.Mixin.define("qx.ui.mobile.container.MNativeScroll",
     _onTrackEnd: function(evt) {
       this._onTrack = false;
 
-      if(this._snapTimeoutId) {
-        clearTimeout(this._snapTimeoutId);
+      if(qx.core.Environment.get("browser.name").indexOf("chrome") == -1 &&
+         qx.core.Environment.get("browser.name").indexOf("safari") == -1) {
```
Comment (oetiker) [inline framework/source/class/qx/ui/mobile/container/MNativeScroll.js]: this seems a rather odd way of checking the browser ... also a feature test would be preferred

File: framework/source/class/qx/ui/mobile/container/MNativeScroll.js [hunk 3]
```diff
@@ -152,11 +159,20 @@ qx.Mixin.define("qx.ui.mobile.container.MNativeScroll",
...
         var snap = this._scrollProperties.snap;
         if (snap) {
-          this._snapPoints = [];
           var snapTargets = this.getContentElement().querySelectorAll(snap);
-          for (var i = 0; i < snapTargets.length; i++) {
-            var snapPoint = qx.bom.element.Location.getRelative(this._getContentElement(), snapTargets[i], "scroll", "scroll");
-            this._snapPoints.push(snapPoint);
+          this._snapPoints = [];
+          if(qx.core.Environment.get("browser.name").indexOf("chrome") == -1 &&
+             qx.core.Environment.get("browser.name").indexOf("safari") == -1) {
```
Comment (oetiker) [inline framework/source/class/qx/ui/mobile/container/MNativeScroll.js]: DRY

**Review by oetiker — CHANGES_REQUESTED** [review]

**Comment by johannes-nivus** [comment]
@oetiker, @hkollmann 
I will implement a feature test and use a corresponding environment variable.
But just a short question: How do I change a PR? Just editing my fork? Close and recommit?

**Comment by hkollmann** [comment]
Just commit to your fork and branch. The rest is done by github

**Comment by hkollmann** [comment]
BTW to held your master branch clean it's easier to create a branch before committing.

**Comment by cboulanger** [comment]
@johannes-nivus We're closing this PR because it is incompatible with current (v7) master. Since we're interested in what the PR is trying to achieve, we would like to ask you to open an new one shoud you upgrade to current master for your development (one major change is that `/framework/source` has been moved to `/source`). Thank you.

**Pull request closed without merging**